config/routes.rb

```diff
@@ -1,7 +1,11 @@
 Rails.application.routes.draw do
 
   get 'home', to: 'home#index'
-  resources :cards, except: [ :show ]
+  resources :cards, except: [ :show ] do
+    member do
+      post 'compare'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.